### PR TITLE
fix: fix fees calculation and add support for L1 fees for Optimism transactions

### LINKED ISSUE #18780: Incorrect Optimism Gas Fee display
### Steps:
1. Recover the user with tokens available on Optimism.
2. Enter a transaction amount on the "send assets" screen.
3. Check the generated fees displayed by the Status App.

### Actual result:
The displayed gas fee is consistently less than $0.01 USD for transactions on the Optimism network. However, upon checking the actual transaction fees on Etherscan, they are substantially higher:
image

Transaction 1 (0.0001 ETH): Actual fee on Etherscan is $0.25 USD.
https://optimistic.etherscan.io/tx/0xdbb244478ec6e4e399c3d62ad065ae3bf06722bb53c533c81da8afb277e1befe

Transaction 2 (0.0005 ETH): Actual fee on Etherscan is $0.26 USD.
https://optimistic.etherscan.io/tx/0x6710f078abda124678357d14bb70e34dae2e03da78abaabbdc7ccf7dcbd6c749

### Expected result:
The fee corresponds to the actual fee shown in the etherscan

### LINKED ISSUE #19431: Fees always displayed as 0 for all networks on the send assets page
### Steps:
1. Recover user with some assets.
2. Go to send assets page.
3. Enter available assets value.
4. Check fees.

### Actual result:
The fees are always shown as 0 for all networks.

image

### Expected result:
The fees should be calculated and displayed for each network.
Here is a screenshot of using MetaMask to estimate the fee for the main network.
image

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #18780 
fixes #19431

### Summary

This PR fixes fees calculation and adds support for Optimism L1 fees.

cc @FFFra 

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

#### General fees fix testing
- Open Status
- Login
- Go to wallet
- Go to an account with funds on any chain
- Tap on Send button
- Select an account to send funds to
- Select token
- Input an amount
- Wait for routes to be loaded
- Check that fee is correctly displayed

#### For Optimism specific L1 fees
- Open Status
- Login
- Go to wallet
- Go to an account with funds on Optimism
- Tap on Send button
- Select an account to send funds to
- Select token
- Input an amount
- Wait for routes to be loaded
- Check that fee is correctly displayed

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/send/input_amount/view.cljs [hunk 1]
```diff
@@ -243,8 +243,9 @@
                                           (<= input-num-value 0)
                                           (> input-num-value (current-limit)))
             amount-text               (str @input-value " " token-symbol)
+            first-route               (first route)
```
Comment (briansztamfater) [inline src/status_im/contexts/wallet/send/input_amount/view.cljs]: We use the first route to identify the native token symbol, which is the token used to pay the fees. This should always be the same token (ETH) across all routes as long as we're only supporting Layer 2s and the mainnet, and not including sidechains or alternative chains. Therefore, selecting the first element as the basis for this information should be entirely safe.

File: src/status_im/contexts/wallet/common/utils/send.cljs [hunk 2]
```diff
@@ -3,20 +3,16 @@
...
-        (money/with-precision (money/div total-gas-fee billion) 10)))))
+  (let [gas-amount         (money/bignumber (get data :gas-amount))
+        gas-fees           (get data :gas-fees)
+        eip1559-enabled?   (get gas-fees :eip-1559-enabled)
+        optimal-price-gwei (money/bignumber (if eip1559-enabled?
+                                              (get gas-fees :max-fee-per-gas-medium)
+                                              (get gas-fees :gas-price)))
+        total-gas-fee-wei  (money/mul (money/->wei :gwei optimal-price-gwei) gas-amount)
+        l1-fee-wei         (money/->wei :gwei (get gas-fees :l-1-gas-fee))]
+    (money/add total-gas-fee-wei l1-fee-wei)))
```
Comment (briansztamfater) [inline src/status_im/contexts/wallet/common/utils/send.cljs:15]: Used the same logic from status-desktop, values are received in gwei and we transform them to wei.

File: src/status_im/contexts/wallet/common/utils/send.cljs [hunk 3]
```diff
@@ -3,20 +3,16 @@
...
 
 (defn calculate-full-route-gas-fee
   [route]
-  (reduce money/add (map calculate-gas-fee route)))
+  (money/wei->ether (reduce money/add (map calculate-gas-fee route))))
```
Comment (briansztamfater) [inline src/status_im/contexts/wallet/common/utils/send.cljs:20]: Sums all the routes fees in wei and then convert the total value to ether

File: src/status_im/contexts/wallet/common/utils/send.cljs [hunk 4]
```diff
@@ -3,20 +3,16 @@
...
-            total-gas-fee (money/mul gas-amount gas-price)]
-        (money/with-precision (money/div total-gas-fee billion) 10)))))
+  (let [gas-amount         (money/bignumber (get data :gas-amount))
+        gas-fees           (get data :gas-fees)
+        eip1559-enabled?   (get gas-fees :eip-1559-enabled)
+        optimal-price-gwei (money/bignumber (if eip1559-enabled?
+                                              (get gas-fees :max-fee-per-gas-medium)
+                                              (get gas-fees :gas-price)))
+        total-gas-fee-wei  (money/mul (money/->wei :gwei optimal-price-gwei) gas-amount)
+        l1-fee-wei         (money/->wei :gwei (get gas-fees :l-1-gas-fee))]
```
Comment (briansztamfater) [inline src/status_im/contexts/wallet/common/utils/send.cljs:14]: `l-1-gas-fee` should be 0 unless it is an Optimism (or Optimism Sepolia) transaction.

For more context: https://docs.optimism.io/stack/transactions/fees#l1-data-fee

**Review by FFFra — APPROVED** [review]

File: src/status_im/contexts/wallet/common/utils/send.cljs (hunk 3 above)
Comment (OmarBasem) [reply src/status_im/contexts/wallet/common/utils/send.cljs:20]: I think it maybe worth it adding that comment "Sums all the routes fees in wei and then convert the total value to ether" as a docstring to the function

**Review by OmarBasem — APPROVED** [review]

File: src/status_im/contexts/wallet/common/utils/send.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/common/utils/send.cljs:20]: Thanks, done!

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater thank you for PR. I think the PR can be merged since all information related to gas fees fetches is implemented in status go. However, please address any issues if you see that the problem is on your side.

I have created a comparison table with gas fees using Metamask:

1 | Mainnet Mainnet | eth | 0.0001 | 0.83 usd | 0.74 usd
-- | -- | -- | -- | -- | --
2 | Mainnet Mainnet | snt | 1 | 5 usd | 4.2 usd
3 | Mainnet Mainnet | uni | 0.01 | 2.33 usd | 3.05 usd
4 | Mainnet Arbitrium | eth | 0.0001 | 0.3 usd | <0.01 usd, but in fact 0.3 usd
5 | Mainnet Arbitrium | wrapped bitcoin | 0.00001 | 0.62 usd | <0.01 usd
6 | Mainnet Optimism | eth | 0.0001 | 0,31 usd | 1.17 usd, in fact 0.30 usd [link](https://optimistic.etherscan.io/tx/0xd2977ec93278a3bcf430fad3d620433f6c1e93609da9eb62a9be3afdc1b7d41a) in fact the fee is in 75% less
7 | Mainnet Optimism | wrapped bitcoin | 0.00001 | 0.62 usd | 1.32 usd is shown

In conclusion, regarding above the table:
- For mainnet transactions (rows 1-3) I am not sure if this is expected behavior that all status app fees 15% less
- It seems like Arbitrum is always shown as 0.01 USD (in row 4 and 5), but in fact, when the transaction is performed within the Status app, the gas fee is the same as within Metamask. So, the actual gas fee is 300-600% higher than estimated. Which is definitely that is not ok
- In Optimism, the gas fee is usually higher than estimated. Status gas fee typically shows as 1.30 USD, but in fact 0.30 usd (in case when working with assets such as ETH) that is not ok
**How gas fee is estimated in Optimism in Metamask**:
image

**How gas fee is estimated in Optimism in Status**:
image

**Comment by VolodLytvynenko** [comment]
### Question
Should the gas fee be fetched on the confirmation page which is shown when collectibles or tokens are going to be sent? Currently, it doesn't
image

**Comment by briansztamfater** [comment]
Hey @VolodLytvynenko, thanks for testing! We have an issue for that: https://github.com/status-im/status-mobile/issues/18408 👍

**Comment by briansztamfater** [comment]
@VolodLytvynenko quick question, do you experience same fees values when trying those transactions in Status Desktop?

**Comment by VolodLytvynenko** [comment]
> @VolodLytvynenko quick question, do you experience same fees values when trying those transactions in Status Desktop?

@briansztamfater hey. Unfortunately can't check it on a desktop. All transactions related to L2 just don't work there, routes are not fetched. In case of L1 the fees are the same as on mobile

**Comment by briansztamfater** [comment]
> > @VolodLytvynenko quick question, do you experience same fees values when trying those transactions in Status Desktop?
> 
> @briansztamfater hey. Unfortunately can't check it on a desktop. All transactions related to L2 just don't work there, routes are not fetched. In case of L1 the fees are the same as on mobile

Ok, no problem, I will merge this so we can have at least some fees displayed, if there are some things to be adjusted, probably will be adjusted on status-go, we can fill another issue if necessary for mobile in the future.
